app/models/link.rb

```diff
@@ -0,0 +1,10 @@
+class Link < ApplicationRecord
+  belongs_to :paper
+  validates_presence_of :url
+  validate :valid_url
```

Comment:
I'm pretty sure you can do both of the validations with one call to validates. Also, did you try using the regex I mentioned in #102?